homeassistant/components/sensibo/climate.py

```diff
@@ -273,68 +280,39 @@ async def async_set_temperature(self, **kwargs) -> None:
             else:
                 return
 
-        result = await self._async_set_ac_state_property(
-            "targetTemperature", int(temperature)
-        )
-        if result:
-            self.coordinator.data[self.unique_id]["target_temp"] = int(temperature)
-            self.async_write_ha_state()
+        await self._async_set_ac_state_property("targetTemperature", int(temperature))
 
     async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set new target fan mode."""
-        result = await self._async_set_ac_state_property("fanLevel", fan_mode)
-        if result:
-            self.coordinator.data[self.unique_id]["fan_mode"] = fan_mode
-            self.async_write_ha_state()
+        await self._async_set_ac_state_property("fanLevel", fan_mode)
 
     async def async_set_hvac_mode(self, hvac_mode: str) -> None:
         """Set new target operation mode."""
         if hvac_mode == HVAC_MODE_OFF:
-            result = await self._async_set_ac_state_property("on", False)
-            if result:
-                self.coordinator.data[self.unique_id]["on"] = False
-                self.async_write_ha_state()
+            await self._async_set_ac_state_property("on", False)
             return
 
         # Turn on if not currently on.
         if not self.coordinator.data[self.unique_id]["on"]:
-            result = await self._async_set_ac_state_property("on", True)
-            if result:
-                self.coordinator.data[self.unique_id]["on"] = True
+            await self._async_set_ac_state_property("on", True)
```

Comment:
Back to my original question: is it ok that this now writes to ha state?

Reply:
I thought about it one more time as yes I think so as this would be more in line with how the device operates.
E.g. to go from cool to off to heat you need to in Sensibo do cool-->off-->on (cool)-->heat.

Reply:
I was just making sure... if you're happy I'm happy...